front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,643 @@
+    const [moduleReferenceError, setModuleReferenceError] = useState<string | undefined>(undefined);
+    const [moduleReferenceLengthError, setModuleReferenceLengthError] = useState<string | undefined>(undefined);
+    const [schemaError, setSchemaError] = useState<string | undefined>(undefined);
+
+    const [isModuleReferenceAlreadyDeployedStep2, setIsModuleReferenceAlreadyDeployedStep2] = useState(false);
+    const [shouldWarnDifferenceModuleReferences, setShouldWarnDifferenceModuleReferences] = useState(false);
+    const [shouldWarnInputParameterInSchemaIgnored, setShouldWarnInputParameterInSchemaIgnored] = useState(false);
+
+    const [txHash, setTxHash] = useState<string | undefined>(undefined);
+
+    const [uploadedModuleSchemaBase64, setUploadedModuleSchemaBase64] = useState<string | undefined>(undefined);
+
+    const [smartContractIndex, setSmartContractIndex] = useState<string | undefined>(undefined);
+    const [inputParameterTemplate, setInputParameterTemplate] = useState<string | undefined>(undefined);
+    const [displayContracts, setDisplayContracts] = useState<string[]>([]);
+
+    // Refresh smartContractIndex periodically.
+    // eslint-disable-next-line consistent-return
+    useEffect(() => {
+        if (connection && client && account && txHash !== undefined) {
+            const interval = setInterval(() => {
+                console.log('refreshing_smartContractIndex');
+                client
+                    .getBlockItemStatus(txHash)
+                    .then((report) => {
+                        if (report !== undefined) {
+                            setSmartContractIndex(undefined);
+                            if (report.status === 'finalized') {
+                                if (
+                                    report.outcome.summary.type === TransactionSummaryType.AccountTransaction &&
+                                    report.outcome.summary.transactionType === TransactionKindString.InitContract
+                                ) {
+                                    setSmartContractIndex(
+                                        report.outcome.summary.contractInitialized.address.index.toString()
+                                    );
+                                    clearInterval(interval);
+                                } else {
+                                    setSmartContractIndexError('Contract initialization failed');
+                                    clearInterval(interval);
+                                }
+                            }
+                        }
+                    })
+                    .catch((e) => {
+                        setSmartContractIndex(undefined);
+                        setSmartContractIndexError((e as Error).message);
+                        clearInterval(interval);
+                    });
+            }, REFRESH_INTERVAL.asMilliseconds());
+        }
+    }, [connection, account, client, txHash]);
```

Comment:
Same as for module deployment, I am not sure why the `account` is needed here (and above in the `if` statement. It is not used.

Reply:
removed